decidim-bulletin_board-js/src/key-ceremony/key-ceremony.test.js

```diff
@@ -1,23 +1,27 @@
-import { tap } from "rxjs/operators";
-import { Subject } from "rxjs";
-
 import {
   KeyCeremony,
   MESSAGE_RECEIVED,
   MESSAGE_PROCESSED,
 } from "./key-ceremony";
 
+import { buildMessageId } from "../test-utils";
+
 jest.mock("../trustee/trustee");
 
 describe("KeyCeremony", () => {
-  const electionLogEntriesUpdates = new Subject();
+  const electionLogEntriesUpdates = [];
 
   const bulletinBoardClient = {
-    getElectionLogEntries: jest.fn(() => Promise.resolve([])),
-    subscribeToElectionLogEntriesUpdates: jest.fn((_args, fn) =>
-      electionLogEntriesUpdates.pipe(tap(fn)).subscribe()
-    ),
-    processKeyCeremonyStep: jest.fn(),
+    getElectionLogEntries: jest.fn(() => {
+      const result = [];
+      for (let i = 0; i < electionLogEntriesUpdates.length; i++) {
+        result.push(electionLogEntriesUpdates.shift());
+      }
```

Comment:
If you just want to copy the values from one array to the other you can do something like this:
```js
const result = [...electionLogEntriesUpdates];
```